packages/lit-dev-content/src/components/litdev-tutorial.ts

```diff
@@ -20,16 +20,26 @@ import {resetIcon} from '../icons/reset-icon.js';
 import {catalogIcon} from '../icons/catalog-icon.js';
 import {forwardArrowIcon} from '../icons/forward-arrow-icon.js';
 import {backArrowIcon} from '../icons/back-arrow-icon.js';
+import {checkFailedIcon} from '../icons/check-failed-icon.js';
+import {checkPassedIcon} from '../icons/check-passed-icon.js';
+import {checkCodeIcon} from '../icons/check-code-icon.js';
+import {loopIcon} from '../icons/loop-icon.js';
 
 import '@material/mwc-icon-button';
+import '@material/mwc-snackbar';
+import {Snackbar} from '@material/mwc-snackbar';
 import './litdev-example-controls.js';
 import './litdev-playground-change-guard.js';
 import './litdev-icon-button.js';
 import {Task, TaskStatus} from '@lit-labs/task';
+import {PostDoc} from 'postdoc-lib';
```

Comment:
It seems a little unfortunate that a bunch of the interesting code that's coming in with this PR is not getting reviewed here, because it was published to a separate postdoc library, and that library wasn't reviewed.

Maybe some background on the problem the library solves would be helpful here.

For the sake of argument, what wouldn't work about this simple approach?:

- Parent sends a "check" message to the iframe.
- iframe receives message, checks `event.source === parentWindow` to confirm the message came from the parent, does the check, and then posts a response back to `parentWindow`
- Parent receives message and checks `event.source === iframe.contentWindow` and does its thing with the response.

Reply:
The issues are around startup time for the iframes and missing the initial handshake messages which complicate the initial message sending. You have to write the same handshake logic on both sides as both need to initiate a handshake and receive a handshake depending on startup.

Additionally, this has to be repeated on reload, but that might be slightly easier with the src changing every time in this case.

Additionally postdoc uses a messagechannel to not pollute the window after handshake is made. This will become more relevant when we implement the "multiple tutorial steps on the same page / scrollable view".

Also with all the repeated code, it makes sense to centralize this into a tiny library, but since the code needs to be both in `/src` and in `/samples/checkable-tutorial.json`, that means the code would have to be duplicated due to our build system, and until we have wireit, there's a lot of cludge around messing with the build system. Hence the simplest solution would be to use something off of NPM.

There are other npm libraries i import and massage into what we need if you prefer. I also don't mind bringing postdoc into the lit org.